File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 15 and 16 June 2019/04. Cinema/04. Cinema.js
function cinema(...arg) {
    let hallPlaces = +arg.shift()
    let allTicketPrice = 0
    let seats = 0

    while (true) {
        let currentPlaces = +arg.shift()

        if (isNaN(currentPlaces)) {
            break
        }

        seats += currentPlaces
        if (seats <= hallPlaces) {
            if (currentPlaces % 3 == 0) {
                currentPlaces--
            }

            allTicketPrice += currentPlaces * 5
        } 
    }

    if (hallPlaces < seats)
        return `The cinema is full.\nCinema income - ${allTicketPrice} lv.`
    return `There are ${hallPlaces - seats} seats left in the cinema.\nCinema income - ${allTicketPrice} lv.`
}

// console.log(cinema(60, 10, 6, 3, 20, 15, 'Movie time!'))
// console.log(cinema(50, 15, 10, 10, 15, 5))

File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 15 and 16 June 2019/04. Cinema/04. Cinema.py
hall_places = int(input())
all_ticket_price = 0
seats_count = 0

while True:
    try:
        current_places = int(input())
        seats_count += current_places
        if seats_count <= hall_places:
            if current_places % 3 == 0:
                current_places -= 1
            all_ticket_price += current_places * 5
        else:
            break
    except:
        print(f'There are {hall_places - seats_count} seats left in the cinema.\nCinema income - {all_ticket_price} lv.')
        exit()


print(f'The cinema is full.\nCinema income - {all_ticket_price} lv.')
